File: service/src/main/java/com/ricardococati/service/util/BigDecimalCustomizado.java
package com.ricardococati.service.util;

import java.math.BigDecimal;
import java.math.RoundingMode;

public class BigDecimalCustomizado {

  public static BigDecimal getDoubleValueBigDecimalHalfUpArredondado4Casas(
      final Double valorDouble) {
    return new BigDecimal(valorDouble).setScale(4, RoundingMode.HALF_UP);
  }

  public static BigDecimal getValueBigDecimalHalfUpArredondado4Casas(
      final BigDecimal valorDouble) {
    return valorDouble.setScale(4, RoundingMode.HALF_UP);
  }

}

File: service/src/main/java/com/ricardococati/service/util/BigDecimalCustomizado.kt
package com.ricardococati.service.util

import java.math.BigDecimal
import java.math.RoundingMode

object BigDecimalCustomizado {
    @JvmStatic
    fun getDoubleValueBigDecimalHalfUpArredondado4Casas(
            valorDouble: Double?): BigDecimal {
        return BigDecimal(valorDouble!!).setScale(4, RoundingMode.HALF_UP)
    }

    @JvmStatic
    fun getValueBigDecimalHalfUpArredondado4Casas(
            valorDouble: BigDecimal): BigDecimal {
        return valorDouble.setScale(4, RoundingMode.HALF_UP)
    }
}